src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1340,49 +1539,69 @@ codet java_bytecode_convert_methodt::convert_instructions(
     {
       assert(op.size()==1 && results.size()==1);
       code_blockt block;
-      // TODO throw NullPointerException instead
-      const typecast_exprt lhs(op[0], pointer_typet(empty_typet()));
-      const exprt rhs(null_pointer_exprt(to_pointer_type(lhs.type())));
-      const exprt not_equal_null(
-        binary_relation_exprt(lhs, ID_notequal, rhs));
-      code_assertt check(not_equal_null);
-      check.add_source_location()
-        .set_comment("Throw null");
-      check.add_source_location()
-        .set_property_class("null-pointer-exception");
-      block.move_to_operands(check);
+      if(throw_runtime_exceptions)
+      {
+        codet null_dereference_check=
+          throw_null_dereference_exception(
+            op[0],
+            i_it->source_location);
+        block.move_to_operands(null_dereference_check);
+      }
+      else
+      {
+        const typecast_exprt lhs(op[0], pointer_typet(empty_typet()));
```

Comment:
Not a suggested change - but I don't really understand what the code looks like if `throw_runtime_exceptions` is disabled - perhaps some documentation (or maybe extract this also into a method so you can document in the name?)